src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T):GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const groupKey = String(item[key]);
+
+    if (!result[groupKey]) {
+      result[groupKey] = items.filter((elem) => groupKey === String(elem[key]));
```

Comment:
do you need to call filter method inside the forEach one?
you've already have a cycle, where you go through all of elements
just check each of them one by one, and add to the needed key